cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/TaskWorkerHttpHandlerInternal.java

```diff
@@ -87,6 +94,20 @@ public void run(FullHttpRequest request, HttpResponder responder) {
     }
   }
 
+  @GET
+  @Path("/token")
+  public void token(io.netty.handler.codec.http.HttpRequest request, HttpResponder responder) {
+    try {
+      URL url = new URL(metadataServiceEndpoint);
```

Comment:
The `metadataServiceEndpoint` can be `null` if it is not configured. You should do a `null` check and returns a `501 Not Implemented`